eodatasets3/model.py

```diff
@@ -535,12 +540,25 @@ def for_s2_derivatives(cls, dataset: EoFields, uri=DEA_URI_PREFIX):
                 "odc:producer",
                 "odc:product_family",
                 "odc:region_code",
-                "odc:dataset_version",
```

Comment:
The derivative naming conventions were added without tests, unfortunately. But I believe they fail without a dataset version because its [used in the folder name](https://github.com/GeoscienceAustralia/eo-datasets/blob/15a199a957998107fa3670fd3d6818f7400712a4/eodatasets3/model.py#L580)

(We have so many variants of naming conventions added by different groups that this is quickly becoming very unwieldly...)